src/main/telemetry/smartport.c

```diff
@@ -831,43 +835,44 @@ void processSmartPortTelemetry(smartPortPayload_t *payload, volatile bool *clear
 #endif
 #ifdef USE_GPS
             case FSSP_DATAID_SPEED      :
-                if (STATE(GPS_FIX)) {
+                if (sendSensorGPS) {
                     //convert to knots: 1cm/s = 0.0194384449 knots
                     //Speed should be sent in knots/1000 (GPS speed is in cm/s)
-                    uint32_t tmpui = gpsSol.groundSpeed * 1944 / 100;
+                    uint32_t tmpui = STATE(GPS_FIX) ? (gpsSol.groundSpeed * 1944 / 100) : 0;
                     smartPortSendPackage(id, tmpui);
                     *clearToSend = false;
                 }
                 break;
             case FSSP_DATAID_LATLONG    :
-                if (STATE(GPS_FIX)) {
-                    uint32_t tmpui = 0;
-                    // the same ID is sent twice, one for longitude, one for latitude
+                if (sendSensorGPS) {
+                    // the same ID is sent twice, alternating for longitude and latitude
                     // the MSB of the sent uint32_t helps FrSky keep track
-                    // the even/odd bit of our counter helps us keep track
-                    if (tableInfo->index & 1) {
-                        tmpui = abs(gpsSol.llh.lon);  // now we have unsigned value and one bit to spare
-                        tmpui = (tmpui + tmpui / 2) / 25 | 0x80000000;  // 6/100 = 1.5/25, division by power of 2 is fast
-                        if (gpsSol.llh.lon < 0) tmpui |= 0x40000000;
+                    int32_t gpsCoordinate = 0;
+                    uint32_t sensorBits = (tableInfo->index & 1) ? 0x80000000 : 0;
+                    if (STATE(GPS_FIX)) {
+                        gpsCoordinate = (tableInfo->index & 1) ? gpsSol.llh.lon : gpsSol.llh.lat;
                     }
-                    else {
-                        tmpui = abs(gpsSol.llh.lat);  // now we have unsigned value and one bit to spare
-                        tmpui = (tmpui + tmpui / 2) / 25;  // 6/100 = 1.5/25, division by power of 2 is fast
-                        if (gpsSol.llh.lat < 0) tmpui |= 0x40000000;
+                    uint32_t tmpui = ABS(gpsCoordinate);
+                    tmpui = (tmpui + tmpui / 2) / 25;  // 6/100 = 1.5/25, division by power of 2 is fast
```

Comment:
Should this just be `tmpui *= 6.0f / 100`?

Reply:
This is from the original code which seems to think that there's a performance reason for doing it this way (div by 2 is more efficient). But it would seem to me that div by 100 should be efficient as well however I don't know the microarchitecture. But it's currently all integer math and I don't suspect there's a need for floats as suggested.